Oracle Daytona does not expose task-bundled skills at /root/.claude/skills, breaking 3d-scan-calc oracle

## Summary

A real Daytona oracle run for `3d-scan-calc` fails because the task solution imports a task-bundled skill from `/root/.claude/skills/mesh-analysis/scripts`, but that path is not populated in the oracle execution environment.

This makes oracle/ground-truth validation fail for at least one SkillsBench task that depends on task-bundled skills.

## Repro

Observed on `v0.5-integration` at `d8fb7af8379e1a71b3e2f170c8e15c64e9fcd026`, SkillsBench `3f86918e2de85e07bf2a6b817a7d00e92aba76c1`.

```bash
export TASKS=.cache/datasets/benchflow-ai/skillsbench/tasks
uv run bench eval create 
  --tasks-dir "$TASKS" 
  --include 3d-scan-calc 
  --agent oracle 
  --sandbox daytona 
  --jobs-dir /tmp/repro-3d-scan-oracle-daytona
```

The continuation audit ran it in a four-task oracle breadth batch; the same task produced this trajectory event:

```json
{
  "type": "oracle",
  "command": "solution/solve.sh",
  "return_code": 1,
  "stdout": "Traceback (most recent call last):
  File "/root/solve_task.py", line 8, in <module>
    from mesh_tool import MeshAnalyzer
ModuleNotFoundError: No module named 'mesh_tool'\n"
}
```

Verifier then failed because `/root/mass_report.json` was never created:

```text
AssertionError: Output file not found
FileNotFoundError: [Errno 2] No such file or directory: "/root/mass_report.json"
```

Evidence path in the audit workspace:

```text
jobs/e2e/20260525-v05-daytona-audit/test-breadth-oracle-daytona/2026-05-25__12-47-29/3d-scan-calc__8ce1d069/trajectory/acp_trajectory.jsonl
```

## Expected

Oracle mode should make task-bundled skills available at the paths expected by the task solution, or run oracle solutions under the same skill-root convention used by the task authoring system.

For this task, `solution/solve.sh` appends:

```python
sys.path.append("/root/.claude/skills/mesh-analysis/scripts")
from mesh_tool import MeshAnalyzer
```

The task includes the skill under:

```text
environment/skills/mesh-analysis/scripts/mesh_tool.py
```

## Why this matters

The `oracle` agent is the ground-truth path used to distinguish BenchFlow/runtime bugs from model failures. If oracle does not get the same task-bundled skill surface as agents, oracle failures become ambiguous and can mask real SkillsBench/BenchFlow regressions.

The same breadth run passed three other oracle Daytona tasks (`powerlifting-coef-calc`, `software-dependency-audit`, `travel-planning`), so this is not a global Daytona outage.

## Suggested fix

Either:

1. During oracle Daytona runs, copy/link task `environment/skills/*` into `/root/.claude/skills/*` before running `solution/solve.sh`; or
2. Run oracle with a documented `HOME`/skill-root contract and update SkillsBench task solutions accordingly.

Add a regression test with a fixture solution that imports a module from a task-bundled skill.

## Found via

2026-05-25 v0.5 Daytona E2E continuation pass, broad oracle coverage for under-tested SkillsBench categories.

--- a/src/benchflow/agents/install.py
+++ b/src/benchflow/agents/install.py
@@ -245,6 +245,13 @@ async def deploy_skills(
             logger.info("Skills already injected via Dockerfile")
             effective_skills = "/skills"
 
+    # Auto-discover task-bundled skills already uploaded to /app/skills
+    # by _start_env_and_upload (from environment/skills/ in the task dir).
+    if not effective_skills and include_task_skills:
+        bundled = task_path / "environment" / "skills"
+        if bundled.is_dir():
+            effective_skills = "/app/skills"
+
     # Distribute to agent-specific discovery paths
     if agent_cfg is not None:
         skill_paths = agent_cfg.skill_paths

--- a/tests/test_agent_setup.py
+++ b/tests/test_agent_setup.py
@@ -332,6 +332,69 @@ async def test_deploy_skills_agent_with_empty_skill_paths_does_not_use_oracle_pa
 
 
 @pytest.mark.asyncio
+async def test_deploy_skills_oracle_autodiscovers_bundled_skills(tmp_path):
+    """Guards the fix from PR #558 for issue #557: oracle mode on Daytona
+    failed for tasks with bundled skills because deploy_skills never detected
+    the environment/skills/ directory that _start_env_and_upload already
+    uploaded to /app/skills.  Without auto-discovery, effective_skills stayed
+    None and the linking step was skipped, so /root/.claude/skills was never
+    populated."""
+    env = MagicMock()
+    env.exec = AsyncMock(return_value=MagicMock(return_code=0, stdout=""))
+    env.upload_dir = AsyncMock()
+
+    task_path = tmp_path / "task"
+    bundled = task_path / "environment" / "skills" / "mesh-analysis" / "scripts"
+    bundled.mkdir(parents=True)
+    (bundled / "mesh_tool.py").write_text("class MeshAnalyzer: pass\n")
+
+    await deploy_skills(
+        env=env,
+        task_path=task_path,
+        skills_dir=None,
+        agent_cfg=None,
+        sandbox_user=None,
+        agent_cwd="/app",
+        task=_make_task(None),
+    )
+
+    env.upload_dir.assert_not_called()
+    env.exec.assert_awaited_once()
+    link_cmd = env.exec.await_args.args[0]
+    assert "ln -sfn /app/skills /root/.claude/skills" in link_cmd
+
+
+@pytest.mark.asyncio
+async def test_deploy_skills_autodiscovery_skipped_when_include_task_skills_false(
+    tmp_path,
+):
+    """Guards the fix from PR #558: auto-discovery must respect
+    include_task_skills=False so self-gen trials start without skills."""
+    env = MagicMock()
+    env.exec = AsyncMock(return_value=MagicMock(return_code=0, stdout=""))
+    env.upload_dir = AsyncMock()
+
+    task_path = tmp_path / "task"
+    bundled = task_path / "environment" / "skills" / "some-skill"
+    bundled.mkdir(parents=True)
+    (bundled / "helper.py").write_text("pass\n")
+
+    await deploy_skills(
+        env=env,
+        task_path=task_path,
+        skills_dir=None,
+        agent_cfg=None,
+        sandbox_user=None,
+        agent_cwd="/app",
+        task=_make_task(None),
+        include_task_skills=False,
+    )
+
+    env.upload_dir.assert_not_called()
+    env.exec.assert_not_called()
+
+
+@pytest.mark.asyncio
 async def test_deploy_skills_raises_when_skill_linking_fails(tmp_path):
     env = MagicMock()
     env.exec = AsyncMock(return_value=MagicMock(return_code=17, stdout="link failed"))

